test/evergreen/code_coverage/parallel_code_coverage.py

```diff
@@ -0,0 +1,202 @@
+    task_list = task_list_info["task_bucket"]
+    verbose = task_list_info['verbose']
+    list_start_time = datetime.now()
+    for task in task_list:
+        if verbose:
+            print("Running task {} in {}".format(task, build_dir))
+
+        start_time = datetime.now()
+        try:
+            os.chdir(build_dir)
+            split_command = task.split()
+            subprocess.run(split_command, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        except subprocess.CalledProcessError as exception:
+            print(f'Command {exception.cmd} failed with error {exception.returncode}')
+        end_time = datetime.now()
+        diff = end_time - start_time
+
+        if verbose:
+            print("Finished task {} in {} : took {} seconds".format(task, build_dir, diff.total_seconds()))
+
+    list_end_time = datetime.now()
+    diff = list_end_time - list_start_time
+
+    if verbose:
+        print("Completed task list in {} : took {} seconds".format(build_dir, diff.total_seconds()))
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--config_path', required=True, help='Path to the json config file')
+    parser.add_argument('-b', '--build_dir_base', required=True, help='Base name for the build directories')
+    parser.add_argument('-j', '--parallel', default=1, type=int, help='How many tests to run in parallel')
+    parser.add_argument('-s', '--setup', action="store_true",
+                        help='Perform setup actions from the config in each build directory')
+    parser.add_argument('-v', '--verbose', action="store_true", help='Be verbose')
+    args = parser.parse_args()
+
+    verbose = args.verbose
+    config_path = args.config_path
+    build_dir_base = args.build_dir_base
+    parallel_tests = args.parallel
+    setup = args.setup
+
+    if verbose:
+        print('Code Coverage')
+        print('=============')
+        print('Configuration:')
+        print('  Config file                      {}'.format(config_path))
+        print('  Base name for build directories: {}'.format(build_dir_base))
+        print('  Number of parallel tests:        {}'.format(parallel_tests))
+        print('  Perform setup actions:           {}'.format(setup))
+
+    if parallel_tests < 1:
+        os.error("Number of parallel tests must be >= 1")
+
+    # Load test config json file
+    with open(config_path) as json_file:
+        config = json.load(json_file)
+
+    if verbose:
+        print('Configuration:')
+        print(config)
+
+    if len(config['test_tasks']) < 1:
+        sys.exit("No test tasks")
+
+    setup_actions = config['setup_actions']
+
+    if setup and len(setup_actions) < 1:
+        sys.exit("No sectup actions")
+
+    if verbose:
+        print('  Setup actions: {}'.format(setup_actions))
+
+    build_dirs = list()
+
+    # Check the relevant build directories exist and have the correct status
+    for build_num in range(parallel_tests):
```

Comment:
Can we break this out into a `check_build_dirs` subfunction?
Similarly for the `build_wiredtiger` and `run_tests` logic below

Reply:
Good suggestion - I'll make the change.